tests/integration-tests/tests/efa/test_efa.py

```diff
@@ -180,47 +224,91 @@ def _test_osu_benchmarks_multiple_bandwidth(
 
 def run_osu_benchmarks(
     mpi_version,
+    benchmark_group,
     benchmark_name,
     partition,
     remote_command_executor,
     scheduler_commands,
+    num_of_instances,
     slots_per_instance,
     test_datadir,
 ):
-    logging.info("Running OSU benchmarks for {0}".format(mpi_version))
+    osu_benchmark_version = "5.7.1"
+    logging.info(f"Running OSU benchmark {osu_benchmark_version}: {benchmark_name} for {mpi_version}")
+
+    # Init OSU benchmarks
+    init_script = _render_jinja_template(
+        template_file_path=test_datadir / "init_osu_benchmarks.sh", osu_benchmark_version=osu_benchmark_version
+    )
     remote_command_executor.run_remote_script(
-        str(test_datadir / "init_osu_benchmarks.sh"),
+        str(init_script),
         args=[mpi_version],
         hide=True,
-        additional_files=[str(test_datadir / "osu-micro-benchmarks-5.6.3.tar.gz")],
+        additional_files=[str(test_datadir / f"osu-micro-benchmarks-{osu_benchmark_version}.tgz")],
+    )
+
+    # Prepare submission script and submit it
+    copyfile(
+        test_datadir / f"osu_{benchmark_group}_submit_{mpi_version}.sh",
+        test_datadir / f"osu_{benchmark_group}_submit_{mpi_version}_{benchmark_name}.sh",
+    )
+    submission_script = _render_jinja_template(
+        template_file_path=test_datadir / f"osu_{benchmark_group}_submit_{mpi_version}_{benchmark_name}.sh",
+        benchmark_name=benchmark_name,
+        osu_benchmark_version=osu_benchmark_version,
+    )
+    result = scheduler_commands.submit_script(
+        str(submission_script),
+        slots=num_of_instances * slots_per_instance,
+        partition=partition,
     )
-    if partition:
-        result = scheduler_commands.submit_script(
-            str(test_datadir / "osu_{0}_submit_{1}.sh".format(benchmark_name, mpi_version)),
-            slots=2 * slots_per_instance,
-            partition=partition,
-        )
-    else:
-        result = scheduler_commands.submit_script(
-            str(test_datadir / "osu_{0}_submit_{1}.sh".format(benchmark_name, mpi_version)),
-            slots=2 * slots_per_instance,
-        )
     job_id = scheduler_commands.assert_job_submitted(result.stdout)
     scheduler_commands.wait_job_completed(job_id)
     scheduler_commands.assert_job_succeeded(job_id)
 
-    output = remote_command_executor.run_remote_command("cat /shared/osu.out").stdout
+    output = remote_command_executor.run_remote_command(f"cat /shared/{benchmark_name}.out").stdout
     return output
 
 
+def _check_osu_benchmarks_results(test_datadir, mpi_version, benchmark_name, output):
+    packet_sizes = [4, 64, 256, 8192, 65536, 524288, 1048576]
```

Comment:
it would be nice to have a comment explaining where these packet sizes come from

Reply:
Changed approach, by comparing all the packet sizes and accepting a max numbers of 3 failures on 24 tests.